### ISSUE TITLE
Creating SchedulableTask in App

### ISSUE THREAD

**Issue by carlospence** [MrCMS/MrCMS#86]
Good day,

I created a SchedulableTask called AnalyseSeatAvailability and anytime the task is about to run and error is thrown. see below.
![image](https://user-images.githubusercontent.com/12647064/46731527-ce138280-cc82-11e8-83f6-9c607ff3464f.png)

If I disable the task and just want pre-existing task to run, another kind of error is thrown. see below
![image](https://user-images.githubusercontent.com/12647064/46731622-0f0b9700-cc83-11e8-84e6-27bf53247bc7.png)

Basically, I created the ScheduableTask in my app and I noticed there is a ninject error always throwing up. I even commented out all my codes in the Task so as to make it bare in other to see if it is my code that is the issue but it is still the same error coming up. I somehow replicated your DeleteExpiredLogsTask.cs. see my code below.

namespace MrCMS.Web.Apps.SyncBus.Tasks
{
    public class AnalyseSeatAvailability : SchedulableTask
    {
        // private readonly IStatelessSession _statelessSession;
        // private readonly ISession _session;

        public AnalyseSeatAvailability()
        {
            // _session = session;
           //  _statelessSession = statelessSession;
        }
        public override int Priority { get { return 0; } }

        protected override void OnExecute()
        {
            var logs = GetLogs();
            CurrentRequestData.ErrorSignal.Raise(new Exception("Executing now" + logs.Count.ToString()));
            //while (logs.Any())
            //{
            //    IList<SeatAvailabilty> currentLogs = logs;
            //    _statelessSession.Transact(session =>
            //    {
            //        foreach (var item in currentLogs)
            //        {
            //            if (item.BookingStage == BookingStage.BookingSuccess)
            //                continue;
            //            var expirydate = item.PresetTime.AddMinutes(item.ExpiryMin);
            //            var diff = DateTime.Compare(expirydate, DateTime.Now);
            //            if (diff < 0)
            //            {
            //                item.IsExpire = true;
            //                item.StageBeforeCancellation = item.BookingStage;
            //                item.BookingStage = BookingStage.SeatCancelled;
            //                var booking =
            //                    _statelessSession.Query<Booking>()
            //                        .Where(i => i.Guid == item.BookingGuid && i.UserGuid == item.UserGuid).Cacheable().SingleOrDefault();
            //               

            //                _statelessSession.Update(item);
            //                _statelessSession.Delete(item);

            //                if (booking != null)
            //                {
            //                    booking.IsCancelled = true;
            //                    _statelessSession.Update(booking);
            //                    EventContext.Instance.Publish<IOnBookingCancelled, BookingCancelledArgs>(new BookingCancelledArgs { Booking = booking });
            //                }
            //            }
            //            //_statelessSession.Delete(log);
            //        }

            //    });
            //    logs = GetLogs();
            //}
        }

        private IList<SeatAvailabilty> GetLogs()
        {
            var result = new List<SeatAvailabilty>();
            //var result = _statelessSession.QueryOver<SeatAvailabilty>()
            //        .Where(data => data.CreatedOn >= CurrentRequestData.Now.AddDays(-1) && data.CreatedOn <= CurrentRequestData.Now)
            //        .Take(1000)
            //        .List();
            CurrentRequestData.ErrorSignal.Raise(new Exception(result.Count.ToString()));
            return result;

        }
    }
}

All the pre-existing task will start working well, when I remove my task completely from the code, delete the task in the TaskSettings table in the DB and finally clear the cache. Immediately I put in my code and the task is re-created in the TaskSettings Table, the error re-occurs.

Can you create a ScheduableTask in a MrCMS app to test and see what am saying. Am currently using version 0.6 of MrCMS

**Comment by Will-Bill** [comment]
Is this local or live?

In the first error, make sure the HTTP_HOST is correct (scroll down on the
error log) - this should be your local system URL set in site settings.

For the second error, in the [TaskSettings] table have a look at the
TypeName column and make sure these match up with your code.

Couple guard clauses to try:

1. In TaskSettingManager.cs change GetAllSettings() to:

        private Dictionary<Type, TaskSettings> GetAllSettings()
        {
            return
_session.QueryOver<TaskSettings>().Cacheable().List().Where(x => x.Type !=
null).ToDictionary(settings => settings.Type);
        }

2. In ScheduledTaskRunner.cs change ExecuteTask to:

        public void ExecuteTask(string type)
        {
            var typeObj = TypeHelper.GetTypeByName(type);
            if (typeObj == null)
            {
                return;
            }
            var schedulableTask = _kernel.Get(typeObj) as SchedulableTask;
            if (schedulableTask == null)
                return;
            try
            {
                SetStatus(typeObj, TaskExecutionStatus.Executing);
                schedulableTask.Execute();
                SetStatus(typeObj, TaskExecutionStatus.Pending,
                    settings => settings.LastCompleted =
CurrentRequestData.Now);
            }
            catch (Exception exception)
            {
                CurrentRequestData.ErrorSignal.Raise(exception);
                SetStatus(typeObj, TaskExecutionStatus.Pending);
            }
        }

====

Step through the code and make sure _taskSettingManager.GetInfo() in
GetPendingScheuledTasks returns the correct URLs.

**Comment by carlospence** [comment]
Thanks it is working now. The HTTP_HOST was the issue. The actual registered site BaseUrl was localhost:9090 but I pointed another Url localhost:9080 so the HTTP_POST was localhost:9080.

But something is quite not clear, prior to me creating my own SchedulableTask, all the previous task where running very well even with the localhost:9080. Am wondering why it all started when I created mine and the problem stops whenever I remove the one I created.

Thanks.
